### PR TITLE
Refactoring tests from PHPUnit to PestPHP

### PR DESCRIPTION
**Pull request by Nelwhix**
I saw Freek's tweet for people who wanted to try out their pest skills. Any advice will be appreciated, Thanks.

### CODE REVIEW

**Comment by freekmurze** [comment]
Could you remove the `.idea` directory from this PR?

**Comment by Nelwhix** [comment]
@freekmurze Done

File: tests/EnumFromTest.php [hunk 1]
```diff
@@ -2,81 +2,54 @@
...
-{
-    /**
-     * @test
-     * @dataProvider unallowedTypes
-     */
-    public function from_will_throw_type_error_for_unallowed_value_types($value)
-    {
-        $this->expectException(TypeError::class);
+test('from resolves all allowed value types', function() {
+    $this->assertTrue(HttpMethod::GET()->equals(HttpMethod::from('GET')));
```
Comment (freekmurze) [inline tests/EnumFromTest.php:16]: Could you convert all assertions to Pest's `expect()` API?

File: tests/EnumFromTest.php [hunk 2]
```diff
@@ -2,81 +2,54 @@
...
-    /** @test */
-    public function try_from_will_result_in_null_values()
-    {
-        $this->assertNull(HttpMethod::tryFrom(''));
-        $this->assertNull(HttpMethod::tryFrom(false));
-        $this->assertNull(HttpMethod::tryFrom(true));
-        $this->assertNull(HttpMethod::tryFrom(1.0));
-        $this->assertNull(HttpMethod::tryFrom(1.4));
-    }
+test('try from will result in null values', function () {
```
Comment (freekmurze) [inline tests/EnumFromTest.php:33]: Use the `it` function instead of `test` where relevant.

File: tests/Pest/AssertSameEnumTest.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+use PHPUnit\Framework\ExpectationFailedException;
+use Spatie\Enum\Phpunit\EnumAssertions;
+use Spatie\Enum\Tests\MyEnum;
+
+it('passes', function () {
+    EnumAssertions::assertSameEnum(MyEnum::A(), MyEnum::A());
+});
+
+it('fails', function () {
+    $this->expectException(ExpectationFailedException::class);
```
Comment (freekmurze) [inline tests/Pest/AssertSameEnumTest.php:14]: Use Pest's `throw` method instead of `expectException`

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: tests/EnumFromTest.php (hunk 1 above)
Comment (Nelwhix) [reply tests/EnumFromTest.php]: Okay done now

File: tests/Pest/AssertSameEnumTest.php (hunk 3 above)
Comment (Nelwhix) [reply tests/Pest/AssertSameEnumTest.php:14]: Okay Done

**Comment by freekmurze** [comment]
These test are failing. Could you update the workflow too to use Pest?

**Comment by Nelwhix** [comment]
Okay done it

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**